locust/runners.py

```diff
@@ -61,65 +63,54 @@ def errors(self):
     def user_count(self):
         return len(self.locusts)
 
-    def weight_locusts(self, amount, stop_timeout = None):
-        """
-        Distributes the amount of locusts for each WebLocust-class according to it's weight
-        returns a list "bucket" with the weighted locusts
-        """
-        bucket = []
-        weight_sum = sum((locust.weight for locust in self.locust_classes if locust.task_set))
-        for locust in self.locust_classes:
+    def filter_true_locust_classes(self, locust_classes):
+        true_locust_classes = []
+        for locust in locust_classes:
             if not locust.task_set:
-                warnings.warn("Notice: Found Locust class (%s) got no task_set. Skipping..." % locust.__name__)
-                continue
+                warnings.warn("Notice: Found Locust class (%s.%s) got no task_set. Skipping..." % (locust.__module__, locust.__name__))
+            else:
+                true_locust_classes.append(locust)
+        return true_locust_classes
 
+    def update_locusts_settings(self, locusts, stop_timeout=None):
+        for locust in self.locusts:
```

Comment:
Should this be `locusts` rather than `self.locusts`? Without this change I get a `LocustError` here when assigning the base URL via the `--host` option.

Reply:
Yeah, this is a bug. I fix it in the recent commit.